src/components/icon/icon.spec.ts

```diff
@@ -0,0 +1,527 @@
+import {
+  it,
+  describe,
+  expect,
+  beforeEach,
+  beforeEachProviders,
+  inject,
+  TestComponentBuilder,
+} from 'angular2/testing';
+import {
+  HTTP_PROVIDERS,
+  Response,
+  ResponseOptions,
+  XHRBackend} from 'angular2/http';
+import {MockBackend} from 'angular2/http/testing';
+import {
+  provide,
+  Component} from 'angular2/core';
+
+import {MdIcon} from './icon';
+import {MdIconRegistry} from './icon-registry';
+
+/** Returns the CSS classes assigned to an element as a sorted array. */
+const sortedClassNames = (elem: Element) => elem.className.split(' ').sort();
+
+/**
+ * Verifies that an element contains a single <svg> child element, and returns that child.
+ */
+const verifyAndGetSingleSvgChild = (element: any): any => {
+  expect(element.childNodes.length).toBe(1);
+  const svgChild = element.childNodes[0];
+  expect(svgChild.tagName.toLowerCase()).toBe('svg');
+  return svgChild;
+};
+
+/**
+ * Verifies that an element contains a single <path> child element whose "id" attribute has
+ * the specified value.
+ */
+const verifyPathChildElement = (element: any, attributeValue: string) => {
+  expect(element.childNodes.length).toBe(1);
+  const pathElement = element.childNodes[0];
+  expect(pathElement.tagName.toLowerCase()).toBe('path');
+  expect(pathElement.getAttribute('id')).toBe(attributeValue);
+};
+
+export function main() {
+  describe('MdIcon', () => {
+
+    beforeEachProviders(() => [
+      MdIconRegistry,
+      HTTP_PROVIDERS,
+      MockBackend,
+      provide(XHRBackend, {useExisting: MockBackend}),
+    ]);
+
+    let builder: TestComponentBuilder;
+    let mdIconRegistry: MdIconRegistry;
+    let httpRequestUrls: string[];
+
+    beforeEach(
+        inject([TestComponentBuilder, MdIconRegistry, MockBackend],
+        (tcb: TestComponentBuilder, mir: MdIconRegistry, mockBackend: MockBackend) => {
+      builder = tcb;
+      mdIconRegistry = mir;
+      // Set fake responses for various SVG URLs.
+      // Keep track of requests so we can verify caching behavior.
+      httpRequestUrls = [];
+      mockBackend.connections.subscribe((connection: any) => {
+        const url = connection.request.url;
+        httpRequestUrls.push(url);
+        switch (url) {
```

Comment:
Can we break these into a separate file like `fake-svgs.ts` or `test-svgs.ts`, where each is associated with its name?  Possibly something like `getFakeSvgHttpResponses`.